### PR TITLE
Revert adding the datalake prefix for list path API in protocol layer…

### PR DESCRIPTION
**Pull request by katmsft**
…'s options.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
What's the motivation of this change?

**Comment by katmsft** [comment]
> What's the motivation of this change?

When generating the protocol layer I found the original change was done during find&replace and is not wanted.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp [hunk 1]
```diff
@@ -760,7 +760,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
           return DeleteParseResult(context, pipeline.Send(context, request));
         }
 
-        struct ListDataLakePathsOptions
+        struct ListPathsOptions
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:763]: Is this exposed to end customers? If so, we need to add it to changelog

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp:763]: It is not.

**Pull request merged**